### PR TITLE
feat: add managed identity authentication to the Azure Service Bus message pump

### LINKED ISSUE #137: Provide support for managed identity authentication for Azure Service Bus message pump
**Is your feature request related to a problem? Please describe.**
Provide support for managed identity authentication for Azure Service Bus message pump. 

This would allow us to use 0 secrets. 

**Describe the solution you'd like**
A new extension that does not require a secret name, but explicitly state it uses Managed Identity.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Provide the capability to authenticate with managed identity (fallback on environment variables) when setting up an Azure Service Bus Queue/Topic message pump.

Closes #137

### CODE REVIEW

File: docs/preview/features/message-pumps/service-bus.md [hunk 1]
```diff
@@ -141,17 +141,26 @@ public class Startup
...
+        services.AddServiceBusTopicMessagePumpWithPrefix(
             "My-Service-Bus-Topic-Name"
             "My-Service-Bus-Subscription-Prefix",
             "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING");
 
+        // Uses managed identity to authenticate with the Service Bus Topic:
+        services.AddServiceBusTopicMessagePumpUsingManagedIdentity(
+            topicName: properties.EntityPath,
+            subscriptionName: "Receive-All", 
+            fullyQualifiedNamespace: "<your-namespace>.servicebus.windows.net"
```
Comment (tomkerkhove) [inline docs/preview/features/message-pumps/service-bus.md:159]: I would personally just pass the namespace name as it's always the same:
```suggestion
            namespaceName: "<your-namespace>"
```

File: src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs [hunk 2]
```diff
@@ -250,6 +246,10 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                 await OpenNewMessageReceiverAsync();
                 await UntilCancelledAsync(stoppingToken);
             }
+            catch (TaskCanceledException)
+            {
+                // Ignore.
```
Comment (tomkerkhove) [inline src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs:251]: Maybe write a trace? I don't know.

File: src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs [hunk 3]
```diff
@@ -419,24 +361,22 @@ public override async Task StopAsync(CancellationToken cancellationToken)
...
+            _loggingScope?.Dispose();
         }
 
         private async Task DeleteTopicSubscriptionAsync(CancellationToken cancellationToken)
         {
-            string rawConnectionString = await GetServiceBusConnectionStringAsync();
-            var serviceBusConnectionProperties = ServiceBusConnectionStringProperties.Parse(rawConnectionString);
-
-            var serviceBusClient = new ServiceBusAdministrationClient(rawConnectionString);
+            ServiceBusAdministrationClient serviceBusClient = await Settings.GetServiceBusAdminClientAsync();
```
Comment (tomkerkhove) [inline src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs:369]: We already do this on https://github.com/arcus-azure/arcus.messaging/pull/178/files#diff-21920603b21d912cc5678a977a854f57051464f5f3eff50f3dccfcd4d50abca5R210-R215, so maybe move this to a method?

File: src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs [hunk 4]
```diff
@@ -105,32 +113,184 @@ private static ServiceBusEntityType ConvertToServiceBusEntityType(ServiceBusEnti
...
+            {
+                // Assign the configured defaults
+                messageHandlerOptions.AutoCompleteMessages = Options.AutoComplete;
+                messageHandlerOptions.MaxConcurrentCalls = Options.MaxConcurrentCalls ?? messageHandlerOptions.MaxConcurrentCalls;
+            }
+
+            return messageHandlerOptions;
+        }
+        
+        private async Task<string> GetConnectionStringAsync()
```
Comment (tomkerkhove) [inline src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs:293]: Thought, should we use these traces?
https://github.com/arcus-azure/arcus.messaging/pull/178/files#diff-21920603b21d912cc5678a977a854f57051464f5f3eff50f3dccfcd4d50abca5L454-L456

File: src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs [hunk 5]
```diff
@@ -105,32 +113,184 @@ private static ServiceBusEntityType ConvertToServiceBusEntityType(ServiceBusEnti
...
+
+        /// <summary>
+        /// Determines the path based on the provided settings where the Azure Service Bus entity is located.
+        /// </summary>
+        internal async Task<string> GetEntityPathAsync()
+        {
+            if (_tokenCredential is null)
+            {
+                string connectionString = await GetConnectionStringAsync();
+                var properties = ServiceBusConnectionStringProperties.Parse(connectionString);
```
Comment (tomkerkhove) [inline src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs:219]: What if the entity path is not part of the connection string?

**Review by tomkerkhove — APPROVED** [review]
LGTM, added a few thoughts

File: src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs (hunk 5 above)
Comment (stijnmoreels) [reply src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs:219]: Good point, thanks! Missed that one.

File: src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs (hunk 4 above)
Comment (stijnmoreels) [reply src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpSettings.cs:293]: Yeah, probably.

File: src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs (hunk 3 above)
Comment (stijnmoreels) [reply src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs:369]: Hmm, I don't think that would be much added value as all the arguments that we would be sending to that method are further up. Adding delegation here would make it less readable. The only duplication here is retrieving the information for the SB client and calling that 'subscription exists' method.

File: src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs:251]: I'll add one.

**Pull request merged**